tuf/store/interfaces.go

```diff
@@ -12,7 +17,8 @@ type MetadataStore interface {
 
 // PublicKeyStore must be implemented by a key service
 type PublicKeyStore interface {
-	GetKey(role string) ([]byte, error)
+	GetKey(role string) (data.PublicKey, error)
+	RotateKey(role string, cs signed.CryptoService, roots ...data.PublicKey) (data.PublicKey, error)
```

Comment:
If we were using GRPC there would have to be translation between the wire types (GRPC would require us to define our own protos), and the TUF types. The manually created part of the server and signer code would do that translation and sits a level higher than the "store" code.

So to answer you first question, I would definitely be happier to see something wrapped around the HTTPStore that could do the signing/validation.